rts/motoko-rts/src/gc/generational.rs

```diff
@@ -118,7 +118,7 @@ static mut OLD_GENERATION_THRESHOLD: usize = 32 * 1024 * 1024;
 static mut PASSED_CRITICAL_LIMIT: bool = false;
 
 #[cfg(feature = "ic")]
-const CRITICAL_MEMORY_LIMIT: usize = (4096 - 512) * 1024 * 1024;
+const CRITICAL_MEMORY_LIMIT: usize = (4096 - 512) * 1024 * 1024 - crate::memory::MEMORY_RESERVE;
```

Comment:
I forget - what is the other -512 for?

Reply:
Oh, I see that's when we start doing aggressive GC. Nvm

Reply:
Yes, the `-512` is the critical limit. I lowered it further by the newly introduced memory reserve...